hardware/Tado.cpp

```diff
@@ -22,26 +22,28 @@
 #include "../json/json.h"
 #include "../webserver/Base64.h"
 #include "Tado.h"
-#include <regex>
-#include <initializer_list>
 
-#define round(a) ( int ) ( a + .5 )
+using namespace std;
+
 const int TADO_POLL_INTERVAL = 30;  // The plugin should collect information from the API every n seconds.
-const std::string TADO_API_ENVIRONMENT_URL = "https://my.tado.com/webapp/env.js";
+const string TADO_API_ENVIRONMENT_URL = "https://my.tado.com/webapp/env.js";
 const int TADO_TOKEN_MAXLOOPS = 12;	// Default token validity is 600 seconds before it needs to be refreshed.
 									// Each cycle takes 30-35 seconds, so let's stay a bit on the safe side.
 
 CTado::~CTado(void)
 {
 }
 
-CTado::CTado(const int ID, const std::string &username, const std::string &password)
+CTado::CTado(const int ID, const string &username, const string &password)
 {
 	m_HwdID = ID;
 	m_TadoUsername = username;
 	m_TadoPassword = password;
 
-	_log.Log(LOG_TRACE, "Tado: Started Tado plugin with ID=" + boost::to_string(m_HwdID) + ", username=" + m_TadoUsername);
+	if (_log.isTraceEnabled())
+	{
+		_log.Log(LOG_TRACE, "Tado: Started Tado plugin with ID=" + boost::to_string(m_HwdID) + ", username=" + m_TadoUsername);
```

Comment:
Please use the printf functionality of the log like

`_log.Log(LOG_TRACE, "Tado: Started Tado plugin with ID=%d, username=%s",m_HwdID, m_TadoUsername.c_str());`

Please avoid using boost functions like `boost::to_string`

(also further in the code)

Reply:
Changed all the _log.Log lines to use printf format. 
Replaced string concatenations in the file with this format as well. Thanks for the suggestion!